label_studio/projects/models.py

```diff
@@ -644,16 +658,41 @@ def display_count(count: int, type: str) -> Optional[str]:
     def _label_config_has_changed(self):
         return self.label_config != self.__original_label_config
 
-    def delete_predictions(self):
-        if flag_set(
-            'fflag_perf_back_lsdv_4695_update_prediction_query_to_use_direct_project_relation',
-            user='auto',
-        ):
-            predictions = Prediction.objects.filter(project=self)
-        else:
-            predictions = Prediction.objects.filter(task__project=self)
+    def should_none_model_version(self, model_version):
+        """
+        Returns True if the model version provided matches the object's model version,
+        or no model version is set for the object but model version exists in ML backend.
+        """
+        return self.model_version == model_version or self.ml_backend_in_model_version
+
+    def delete_predictions(self, model_version=None, *args, **kwargs):
+        """
+        Deletes the predictions based on the provided model version.
+        If no model version is provided, it deletes all the predictions for this project.
+
+        :param model_version: Identifier of the model version (default is None)
+        :type model_version: str, optional
+        :return: Dictionary with count of deleted predictions
+        :rtype: dict
+        """
+        params = {'project': self}
+
+        if model_version:
+            params.update({'model_version': model_version})
+
+        predictions = Prediction.objects.filter(**params)
         count = predictions.count()
-        predictions.delete()
+
+        with transaction.atomic():
+            # If we are deleting specific model_version then we need
+            # to remove that from the project
+            if self.should_none_model_version(model_version):
+                self.model_version = None
+                self.save()
+
+            predictions.delete()
```

Comment:
optional: I believe the delete() method returns the number of rows that it deleted, so we can ditch the count() query